### DISCUSSION TITLE
HPA for custom resource

### DISCUSSION THREAD

**Discussion by igorekpotworek** [fluxcd/flux2#3625] [General]
According to [FAQ ](https://fluxcd.io/flux/faq/#what-is-a-kustomization-reconciliation): when using a HorizontalPodAutoscaler for Deployment the `spec.replicas` field should be omitted from the Deployments YAML.
I tried to do the same when using HPA on Custom Resource but it failed with:
```
Internal error occurred: the spec replicas field ".spec.replicas" does not exist
```
When `spec.replicas` is specified then I'm getting constant drift: i.e. HPA is scaling down the resource but then on the next Flux reconciliation resource is scaled up and so on.
Is there any fix for this issue or it's just impossible to use HPA on CRD when using Flux?

**Comment by pstibrany** [comment]
Hi, I'm running into the same problem. I wonder if you've found any solution for this?

**Comment by pstibrany** [comment]
It seems that the solution is to 1) remove `spec.replicas` from definition, 2) flux-ignore the resource, 3) add field manually, 4) [transfer ownership of the field](https://kubernetes.io/docs/reference/using-api/server-side-apply/#transferring-ownership), and 5) remove flux-ignore.

**Comment by belal67** [comment]
I'm facing same issue. any update regarding this?

**Reply by stefanprodan** [reply]
Ignoring specific fields from drift detection is planned but not yet supported. Ref: https://github.com/fluxcd/pkg/issues/696

**Reply by stefanprodan** [reply]
To make this work with Flux, you would need to create a Helm chart and use a Flux HelmRelease to apply the custom resource and HPA. The HelmRelease API supports ignoring fields from drift: https://fluxcd.io/flux/components/helm/helmreleases/#ignore-rules
